packages/core/src/core/executer.ts

```diff
@@ -49,17 +50,45 @@ export class Executer {
         process.on('beforeExit', this.closeChild);
         process.on('exit', this.closeChild);
 
-        child.on('exit', (code) => {
+        child.once('exit', (code) => {
             this.logger.info(`"${name}" PID ${child.pid} ${!code ? 'done' : `exit ${code}`}`);
             this.child = undefined;
         });
     }
 
     async closeChild() {
-        while (this.child && !this.child.killed) {
-            this.child.kill();
-            await delay(50);
+        if (!this.child) {
+            return;
+        }
+
+        const child = this.child;
+
+        const exitPromise = new Promise<void>((resolve) => {
+            child.once('exit', () => {
+                resolve();
+            });
+        });
+
+        try {
+            await this.terminateChild(child);
+            await exitPromise;
```

Comment:
https://github.com/ErKeLost/farmup/blob/fix/child_killed/packages/core/src/core/executer.ts#L74

maybe it's the closeChild called by the exit event, in which case there will be a problem